### PR TITLE
Add support for checkpoint frequency option. Also rename RegisterObse…

### PR DESCRIPTION
**Pull request by mkolt**
…rverAsync(IChangeFeedObserverFactory) to RegisterObserverFactoryAsync.

### CODE REVIEW

**Comment by mkolt** [comment]
@shireeshkthota

File: samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/CheckpointFrequency.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    /// <summary>
+    /// Specifies the frequency of lease event. The event will trigger when either of conditions is satisfied.
+    /// </summary>
+    public class CheckpointFrequency
+    {
+        /// <summary>
+        /// Gets or set the value that specifies to checkpoint every speficied number of docs.
+        /// </summary>
+        public uint? ProcessedDocCount { get; set; }
```
Comment (arramac) [inline samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/CheckpointFrequency.cs]: ProcessedDocumentCount

File: samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/ChangeFeedEventHost.cs [hunk 2]
```diff
@@ -555,5 +609,24 @@ public WorkerData(Task task, IChangeFeedObserver observer, ChangeFeedObserverCon
 
             public CancellationTokenSource Cancellation { get; private set; }
         }
+
+        /// <summary>
+        /// Stats since last checkpoint.
+        /// </summary>
+        private class CheckpointStats
+        {
```
Comment (arramac) [inline samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/ChangeFeedEventHost.cs]: TotalRequestUnits also?

File: samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/CheckpointFrequency.cs (hunk 1 above)
Comment (mkolt) [inline samples/ChangeFeedProcessor/DocumentDB.ChangeFeedProcessor/CheckpointFrequency.cs]: OK, thanks!

**Pull request merged**